screen/send/details/hooks/useCreateTransaction.js

```diff
@@ -0,0 +1,260 @@
+      }
+    }
+
+    try {
+      await createPsbtTransaction();
+    } catch (Err) {
+      setIsLoading(false);
+      Alert.alert(loc.errors.error, Err.message);
+      ReactNativeHapticFeedback.trigger('notificationError', { ignoreAndroidSystemSettings: false });
+    }
+  };
+
+  const createPsbtTransaction = async () => {
+    const changeAddress = await getChangeAddressAsync();
+    const requestedSatPerByte = Number(feeRate);
+    const lutxo = utxo || wallet.getUtxo();
+
+    const targets = [];
+    for (const transaction of addresses) {
+      if (transaction.amount === BitcoinUnit.MAX) {
+        // output with MAX
+        targets.push({ address: transaction.address });
+        continue;
+      }
+      const value = parseInt(transaction.amountInCoins);
+      if (value > 0) {
+        targets.push({ address: transaction.address, value });
+      } else if (transaction.amount) {
+        if (currency.btcToSatoshi(transaction.amount) > 0) {
+          targets.push({ address: transaction.address, value: currency.btcToSatoshi(transaction.amount) });
+        }
+      }
+    }
+
+    const { tx, outputs, psbt, fee } = wallet.createTransaction(lutxo, targets, requestedSatPerByte, changeAddress, isTransactionReplaceable ? HDSegwitBech32Wallet.defaultRBFSequence : HDSegwitBech32Wallet.finalRBFSequence);
+
+    if (tx && routeParams.launchedBy && psbt) {
+      console.warn('navigating back to ', routeParams.launchedBy);
+      navigation.navigate(routeParams.launchedBy, { psbt });
+    }
+
+    if (wallet.type === WatchOnlyWallet.type) {
+      // watch-only wallets with enabled HW wallet support have different flow. we have to show PSBT to user as QR code
+      // so he can scan it and sign it. then we have to scan it back from user (via camera and QR code), and ask
+      // user whether he wants to broadcast it
+      navigation.navigate('PsbtWithHardwareWallet', {
+        memo: transactionMemo,
+        fromWallet: wallet,
+        psbt,
+        launchedBy: routeParams.launchedBy,
+      });
+      setIsLoading(false);
+      return;
+    }
+
+    if (wallet.type === MultisigHDWallet.type) {
+      navigation.navigate('PsbtMultisig', {
+        memo: transactionMemo,
+        psbtBase64: psbt.toBase64(),
+        walletID: wallet.getID(),
+        launchedBy: routeParams.launchedBy,
+      });
+      setIsLoading(false);
+      return;
+    }
+
+    txMetadata[tx.getId()] = {
+      txhex: tx.toHex(),
+      memo: transactionMemo,
+    };
+    await saveToDisk();
+
+    let recipients = outputs.filter(({ address }) => address !== changeAddress);
+
+    if (recipients.length === 0) {
+      // special case. maybe the only destination in this transaction is our own change address..?
+      // (ez can be the case for single-address wallet when doing self-payment for consolidation)
+      recipients = outputs;
+    }
+
+    navigation.navigate('Confirm', {
+      fee: new BigNumber(fee).dividedBy(100000000).toNumber(),
```

Comment:
100000000 might be some const

Reply:
It's refactoring from bluewallet part, I just moved it to hook.